src/client/Microsoft.Identity.Client/MsalError.cs

```diff
@@ -772,6 +772,12 @@ public static class MsalError
         /// </summary>
         public const string CertWithoutPrivateKey = "cert_without_private_key";
 
+        /// <summary>
+        /// <para>What happens?</para>The ADAL cache is invalid as it contains multiple refresh token entries for one user.
+        /// <para>Mitigation</para>Delete the ADAL cache. If you do not maintain an ADAL cache, this may be a bug in MSAL.
```

Comment:
>this may be a bug in MSA [](start = 96, length = 24)

do we know if it's a bug or not? seems like this is something we should be able to test and know for sure...?

Reply:
If you manually edit a cache file, you could end up with this. Otherwise, with the current code, I don't see how you can. But there so many combinations of tokens in ADAL - with client info, without client info, with / without upn etc. etc. etc. It's impossible to know for sure. I am also doubting if I should add that comment about "possible MSAL bug", but the reality is there are bugs in the ADAL fallback.

Reply:
I agree. This is why we need this: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1770